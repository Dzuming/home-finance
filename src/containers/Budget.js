import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import * as budgetActions from '../actions/budgetActions';

class Budget extends Component {
  componentDidMount() {
    this.props.actions.fetchBudget();
  }

  render() {
    const { budget } = this.props;
    return <div>{budget} z≈Ç</div>;
  }
}

Budget.propTypes = {
  actions: PropTypes.shape({
    fetchBudget: PropTypes.Func,
  }),
  budget: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};
const mapStateToProps = state => ({
  budget: state.budget,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(budgetActions, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(Budget);
