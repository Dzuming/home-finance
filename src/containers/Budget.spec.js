import React from 'react';
import { create } from 'react-test-renderer';
import Budget from './Budget';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const store = mockStore({budget: 0, user: [], login: {isAuthenticated: true}});
const component = create(
  <Provider store={store}><Budget/></Provider>
);

describe('<Budget />', () => {
  it('should match snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

