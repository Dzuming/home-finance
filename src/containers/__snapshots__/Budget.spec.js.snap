// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Budget /> should match snapshot 1`] = `
<div
  className="Connect-Budget--budget-1"
>
  0
   z≈Ç
</div>
`;
